### PR TITLE
Removed the LogRecord.inferCaller calls that's plaguing System.out performance

### LINKED ISSUE #7433: Avoid inferring caller in LogRecord.getSourceClassName and getSourceMethodName when processing System.out calls
Currently the code is trying to figure out the JUL caller, even in cases where we are just logging our System.out.print "events".  inferCaller is expensive so we need to stop doing that.

 ```
4XESTACKTRACE at java/lang/J9VMInternals.getStackTrace(Native Method)
 4XESTACKTRACE at java/lang/Throwable.getInternalStackTrace(Throwable.java:284(Compiled Code))
 4XESTACKTRACE at java/lang/Access.getStackTraceDepth(Access.java:117(Compiled Code))
 4XESTACKTRACE at java/util/logging/LogRecord.inferCaller(LogRecord.java:557(Compiled Code))
 4XESTACKTRACE at java/util/logging/LogRecord.getSourceClassName(LogRecord.java:319(Compiled Code))
 4XESTACKTRACE at com/ibm/ws/logging/internal/impl/BaseTraceFormatter.messageLogFormat(BaseTraceFormatter.java:462(Compiled Code))
 4XESTACKTRACE at com/ibm/ws/logging/internal/impl/BaseTraceService.invokeMessageRouters(BaseTraceService.java:712(Compiled Code))
 4XESTACKTRACE at com/ibm/ws/logging/internal/impl/BaseTraceService.echo(BaseTraceService.java:644(Compiled Code))
 4XESTACKTRACE at com/ibm/ws/logging/internal/impl/BaseTraceService$TrOutputStream.flush(BaseTraceService.java:1248(Compiled Code)) 5XESTACKTRACE (entered lock: com/ibm/ws/logging/internal/impl/BaseTraceService$TrOutputStream@0x00000000E02E4150, entry count: 1)
 4XESTACKTRACE at java/io/PrintStream.write(PrintStream.java:506(Compiled Code)) 5XESTACKTRACE (entered lock: com/ibm/ws/logging/internal/impl/BaseTraceService$TeePrintStream@0x00000000E02D5EC8, entry count: 1)
```

### PR DESCRIPTION
**Pull request by fwji**
LogRecord.getSourceClassName() and LogRecord.getSourceMethodName(). This
will improve the overall sysout performance by 300%

fixes #7433

### CODE REVIEW

**Comment by fwji** [comment]
#build

**Comment by fwji** [comment]
#build

**Comment by fwji** [comment]
#build

**Comment by fwji** [comment]
#build

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/source/LogSource.java [hunk 1]
```diff
@@ -130,8 +131,14 @@ public LogTraceData parse(RoutedMessage routedMessage, LogRecord logRecord) {
         logData.setModule(logRecord.getLoggerName());
         logData.setSeverity(LogFormatUtils.mapLevelToType(logRecord));
         logData.setLoglevel(LogFormatUtils.mapLevelToRawType(logRecord));
-        logData.setMethodName(logRecord.getSourceMethodName());
-        logData.setClassName(logRecord.getSourceClassName());
+
+        if (logRecord.getLoggerName() != null & logRecord.getLoggerName().equals(SYSOUT)) {
+            logData.setMethodName("");
```
Comment (donbourne) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/source/LogSource.java]: should this not also watch for SystemErr?

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java [hunk 2]
```diff
@@ -459,7 +459,7 @@ public String messageLogFormat(LogRecord logRecord, String formattedVerboseMsg)
         // This is a very light trace format, based on enhanced:
         StringBuilder sb = new StringBuilder(256);
         String sym = getMarker(logRecord);
-        String name = nonNullString(logRecord.getLoggerName(), logRecord.getSourceClassName());
+        String name = nonNullString(logRecord.getLoggerName(), null);
```
Comment (donbourne) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java]: wouldn't this affect System.out / System.err requests too?

**Comment by fwji** [comment]
#build

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/source/LogSource.java (hunk 1 above)
Comment (fwji) [reply dev/com.ibm.ws.logging/src/com/ibm/ws/logging/source/LogSource.java]: Added SystemErr handling

**Comment by fwji** [comment]
#build

File: dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java [hunk 3]
```diff
@@ -459,7 +461,12 @@ public String messageLogFormat(LogRecord logRecord, String formattedVerboseMsg)
         // This is a very light trace format, based on enhanced:
         StringBuilder sb = new StringBuilder(256);
         String sym = getMarker(logRecord);
-        String name = nonNullString(logRecord.getLoggerName(), logRecord.getSourceClassName());
+        String name = null;
+        if (logRecord.getLoggerName() != null & (logRecord.getLoggerName().equals(SYSOUT) ||
+                                                 logRecord.getLoggerName().equals(SYSERR)))
+            name = nonNullString(logRecord.getLoggerName(), null);
```
Comment (donbourne) [inline dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java]: you already know getLoggeName != null here.  could just say name=logRecord.getLoggerName()

**Comment by fwji** [comment]
#build

**Pull request merged**